rfcs/SubschemaComposability.md

```diff
@@ -0,0 +1,239 @@
+terms will be avoided by this RFC until consensus is achieved on a specific
+meaning that enhances the lexicon: "schema merging," "schema stitching," "schema
+federation," "type federation, "type merging."
+
+Additionally, the below terms are uses as follows:
+
+## `Overlapping`
+
+The term `overlapping` will be used to refer to a situation where different
+subschemas define GraphQL elements (such as types, fields, arguments, etc) that
+each map to the same entity (type, field, argument) etc. in the composite
+schema. In the naïve situation, these overlapping elements are known to be
+overlapping simply because they have the same name. However, as mentioned during
+our first WG, a method of composition may be devised in which elements in
+different subschemas with different names are mapped to the same element in the
+composite schema via some set of guiding metadata.
+
+## `Composability`
+
+For a given set of subschemas, it may be possible to generate a valid
+`composite schema` that contains all GraphQL elements of all of the individual
+subschemas. This set of subchemas is then considered to be `composable`.
+Inversely, it may be impossible to generate a valid composite schema containing
+all of the GraphQL elements of a given set of subschemas; this set of subschemas
+is considered `not composable`.
+
+For a given set of subschemas and a given set of overlapping GraphQL elements
+(e.g. types, fields, arguments, etc.), it may be possible to generate a valid
+`composite` GraphQL element containing all of the overlapping GraphQL elements'
+constituent GraphQL sub-element across all of the individual subschemas. This
+GraphQL element is `composable` across the given set of overlapping elements. A
+set of subschemas is `composable` if and only if each of its overlapping GraphQL
+elements are `composable`.
+
+# Types of Composition
+
+Many types of schema composition exist, with potential distinctions including
+(but not limited to):
+
+## Timing of Composition
+
+1. "Build-time" composition in which the composite schema is created during a
+   build process prior to system start-up.
+2. "Run-time" composition in which the composite schema is created dynamically
+   at system start-up.
+3. "Dynamic" composition , in that subschemas are polled by a managing service
+   (or report changes to a managing service), such that the composition process
+   can be triggered at or after system start-up and that the composite schema is
+   dynamically refreshed as the subschemas change (or go offline!).
+
+## Method of Composite Schema Execution
+
+1. `Subschemas as remote GraphQL services`: in certain forms of composition
+   (gateways, gateway schemas, etc.), the execution of a request against a
+   composite schema will yield request(s) against spec-compliant (or even
+   non-spec compliant) GraphQL subservices that are then merged into a single
+   request.
+2. `Single-service composite schema execution`: Often, especially when utilizing
```

Comment:
I think this needs a bit of clarification. For instance if one of the services has a graphql endpoint, then all requests would be formatted as graphql requests. In such case this case wouldn't be different from the first case.
Maybe this case could be better understood if it has "non-GrapgQL services" somewhere.